### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by josepetew**
Added NextJs setup for the frontend. We now use `npm run export` to bundle our static files and then in Django we reference to those files and serve them. The frontend gets the data via REST using the GET sequence endpoint.
There are 2 wins with this setup.
- We have hot reloading in the front end dev environment.
- We pre-render the frontend using NEXT.js. (good for SEO)
- Simpler build process.

Some updates are required in order to deploy to render. Details how to build are in the Readme file.
Talk soon 👍

### CODE REVIEW

**Comment by josepetew** [comment]
Screenshot 2023-03-24 at 21 45 11

File: frontend/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
+
+## Getting Started
```
Comment (franciscojavierarceo) [inline frontend/README.md:3]: can you specify versions of yarn and npm?

File: README.md [hunk 2]
```diff
@@ -62,17 +62,16 @@ https://wordlejj.onrender.com/
...
+npm i
 npm run dev
-python manage.py collectstatic
 ```
 To compile new JS you just have to run
 ```
-npm run dev
-python manage.py collectstatic
+npm run export
 ```
```
Comment (franciscojavierarceo) [inline README.md:75]: oh oh i see here

**Review by franciscojavierarceo — APPROVED** [review]
this looks good to me. to confirm, django is still serving the content, yeah? it looks like it but wanted to verify.

**Comment by josepetew** [comment]
> this looks good to me. to confirm, django is still serving the content, yeah? it looks like it but wanted to verify.

Correct, Django serves the statically generated frontend.

**Comment by franciscojavierarceo** [comment]
nice i'll try this locally

File: README.md [hunk 3]
```diff
@@ -62,17 +62,16 @@ https://wordlejj.onrender.com/
 The FE was developed largely based on SaaS Pegasus' guide on React and Django
 available [here](https://www.saaspegasus.com/guides/modern-javascript-for-django-developers/).
 
+NOTE: Make sure django server is running before running the frontend.
+
 To get things behaving you simply have to run:
 ```
-npm init -y
-npm install webpack webpack-cli --save-dev
+npm i
```
Comment (franciscojavierarceo) [inline README.md:69]: hmm i got this from pulling down the latest changes.

```
[main]% npm i
npm notice 
npm notice New major version of npm available! 8.19.3 -> 9.6.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.6.2
npm notice Run npm install -g npm@9.6.2 to update!
npm notice 
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /Users/franciscojavierarceo/GitHub/wordle-jiu-jitsu/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/Users/franciscojavierarceo/GitHub/wordle-jiu-jitsu/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/franciscojavierarceo/.npm/_logs/2023-03-25T09_33_23_595Z-debug-0.log
```

Comment (franciscojavierarceo) [reply README.md:69]: 🤔

Comment (franciscojavierarceo) [reply README.md:69]: it worked under `./frontend` though

Comment (josepetew) [reply README.md:69]: Will add it to the docs 👍. Indeed its needed to cd into /frontend

**Pull request merged**